src/Features/VisualBasic/Portable/Structure/InternalStructureBlockStructureProvider.vb

```diff
@@ -0,0 +1,290 @@
+            Return BlockHas(NoStatements:=False, block, StartingAt, NextSection, Statements, BannerText)
+        End Function
+
+        Friend Function BlockHas(NoStatements As Boolean, block As SyntaxNode, StartingAt As Object, NextSection As SyntaxNode, Statements As SyntaxList(Of StatementSyntax), BannerText As String) As BlockSpan?
+            If StartingAt IsNot Nothing Then
+                Dim MinLineDelta = 0
+                Dim FinishingAt As SyntaxTrivia?
+                If NoStatements Then
+                    FinishingAt = LastEndOfLineOrNullable(NextSection.GetLeadingTrivia)
+                    MinLineDelta = 1
+                Else
+                    FinishingAt = If(LastEndOfLineOrNullable(NextSection.GetLeadingTrivia),
+                                        LastEndOfLineOrNullable(Statements.Last.GetTrailingTrivia))
+                    MinLineDelta = 0
+                End If
+                If FinishingAt IsNot Nothing Then
+                    If TypeOf StartingAt Is SyntaxTrivia? Then
+                        Return MakeIfHasLineDeltaGreaterThanX(MinLineDelta, DirectCast(StartingAt, SyntaxTrivia?).Value, FinishingAt.Value, BannerText)
+                    ElseIf TypeOf StartingAt Is SyntaxNode Then
+                        Return MakeIfHasLineDeltaGreaterThanX(MinLineDelta, DirectCast(StartingAt, SyntaxNode), FinishingAt.Value, BannerText)
+                    End If
+                End If
+            End If
+            Return Nothing
+        End Function
+
+        Private Function MakeBlockSpan(Start As Integer, Finish As Integer, BannerText As String) As BlockSpan?
+            Dim Span = Text.TextSpan.FromBounds(Start, Finish)
+            Return New BlockSpan(BlockTypes.Statement, True, Span, BannerText)
+        End Function
+
+#Region "MakeIfHasLineDeltaGreaterThanX Overloads"
+        Private Function MakeIfHasLineDeltaGreaterThanX(X As Integer, StartingAt As SyntaxTrivia, FinishingAt As SyntaxTrivia, BannerText As String) As BlockSpan?
+            Return If(LineDelta(StartingAt, FinishingAt) > X, MakeBlockSpan(StartingAt.SpanStart, FinishingAt.SpanStart, BannerText), Nothing)
+        End Function
+
+        Private Function MakeIfHasAtLeastOneLineDelta(X As Integer, StartingAt As SyntaxTrivia, FinishingAt As SyntaxNode, BannerText As String) As BlockSpan?
+            Return If(LineDelta(StartingAt, FinishingAt) > X, MakeBlockSpan(StartingAt.SpanStart, FinishingAt.SpanStart, BannerText), Nothing)
+        End Function
+
+        Private Function MakeIfHasLineDeltaGreaterThanX(X As Integer, StartingAt As SyntaxNode, FinishingAt As SyntaxTrivia, BannerText As String) As BlockSpan?
+            Return If(LineDelta(StartingAt, FinishingAt) > X, MakeBlockSpan(StartingAt.SpanStart, FinishingAt.SpanStart, BannerText), Nothing)
+        End Function
+
+        Private Function MakeIfHasAtLeastOneLineDelta(X As Integer, StartingAt As SyntaxNode, FinishingAt As SyntaxNode, BannerText As String) As BlockSpan?
```

Comment:
whare are these functions for?  doesn't the outlining system already not create outlining it it's not at least one line tall?